src/community/sldService/src/main/java/org/geoserver/sldservice/rest/ClassifierController.java

```diff
@@ -0,0 +1,443 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.xml.transform.TransformerException;
+
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.catalog.LayerInfo;
+import org.geoserver.catalog.ResourceInfo;
+import org.geoserver.config.util.XStreamPersister;
+import org.geoserver.rest.ResourceNotFoundException;
+import org.geoserver.rest.RestBaseController;
+import org.geoserver.rest.catalog.AbstractCatalogController;
+import org.geoserver.rest.converters.XStreamMessageConverter;
+import org.geoserver.sldservice.utils.classifier.ColorRamp;
+import org.geoserver.sldservice.utils.classifier.RulesBuilder;
+import org.geoserver.sldservice.utils.classifier.impl.BlueColorRamp;
+import org.geoserver.sldservice.utils.classifier.impl.CustomColorRamp;
+import org.geoserver.sldservice.utils.classifier.impl.GrayColorRamp;
+import org.geoserver.sldservice.utils.classifier.impl.JetColorRamp;
+import org.geoserver.sldservice.utils.classifier.impl.RandomColorRamp;
+import org.geoserver.sldservice.utils.classifier.impl.RedColorRamp;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.styling.Rule;
+import org.geotools.styling.SLDTransformer;
+import org.geotools.util.NullProgressListener;
+import org.geotools.util.logging.Logging;
+import org.opengis.feature.type.FeatureType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.converters.Converter;
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.MultiLineString;
+import com.vividsolutions.jts.geom.MultiPoint;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.Polygon;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import net.sf.json.xml.XMLSerializer;
+
+/**
+ * ClassifierController.
+ */
+@RestController
+@ControllerAdvice
+@RequestMapping(path = RestBaseController.ROOT_PATH + "/sldservice")
+public class ClassifierController extends AbstractCatalogController {
```

Comment:
Formatted with tabs. Please check all files that have been basically written anew, as well as any new code inline with existing files, and make sure the formatting is correct. Being a community module and considering the REST port basically broke any attempt at backporting changes easily and track history of changes, a mass reformat of the whole module would not be a bad idea imho.

Reply:
Done